### PR TITLE
Add test for loading corrupted metadata

### PR DESCRIPTION
**Pull request by katlapinka**
Original PR: https://github.com/Open-CAS/open-cas-linux/pull/716
Needs: https://github.com/Open-CAS/open-cas-linux/pull/1523

### CODE REVIEW

File: test/functional/tests/security/test_load_corrupted_metadata.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            create=True
+        )
+        core_dev_link = Symlink.get_symlink(
+            link_path="/dev/disk/by-id/core_dev",
+            target="/dev/ram1",
+            create=True
+        )
+
+        cache_dev = Device(cache_dev_link.full_path)
+        core_dev = Device(core_dev_link.full_path)
```
Comment (Deixx) [inline test/functional/tests/security/test_load_corrupted_metadata.py:69]: use RamDisk

File: test/functional/tests/security/test_load_corrupted_metadata.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                    cache.stop()
+                    break
+
+            with TestRun.step("Stop cache."):
+                cache.stop()
+
+    with TestRun.step("Delete symlinks and unload 'brd' module."):
+        remove(cache_dev.path, True)
+        remove(core_dev.path, True)
+        unload_kernel_module('brd')
```
Comment (Deixx) [inline test/functional/tests/security/test_load_corrupted_metadata.py:127]: use Ramdisk

**Review by Deixx — DISMISSED** [review]

File: test/functional/tests/security/test_load_corrupted_metadata.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+@pytest.mark.parametrizex("cache_line_size", CacheLineSize)
+@pytest.mark.parametrizex("cleaning_policy", CleaningPolicy)
+@pytest.mark.parametrizex("promotion_policy", PromotionPolicy)
+@pytest.mark.parametrizex("seq_cutoff_policy", SeqCutOffPolicy)
+def test_load_corrupted_metadata(cache_mode, cache_line_size, cleaning_policy,
+                                 promotion_policy, seq_cutoff_policy):
+    """
+        title: Security test for loading cache with corrupted metadata.
+        description: |
+          Validate the ability of Open CAS to load cache instance and work if metadata is corrupted.
```
Comment (Kamoppl) [inline test/functional/tests/security/test_load_corrupted_metadata.py:36]: `            Validate the ability of Open CAS to load cache instance and work if metadata is corrupted.`

File: test/functional/tests/security/test_load_corrupted_metadata.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            cache = casadm.start_cache(cache_dev, cache_mode, cache_line_size, force=True)
+            core = casadm.add_core(cache, core_dev)
+            metadata_size = int(cache.metadata_size_on_disk.get_value())
+
+        with TestRun.step("Configure cache."):
+            cache.set_cleaning_policy(cleaning_policy)
+            cache.set_promotion_policy(promotion_policy)
+            cache.set_seq_cutoff_policy(seq_cutoff_policy)
+
+        with TestRun.step("Fill cache with random data."):
```
Comment (Deixx) [inline test/functional/tests/security/test_load_corrupted_metadata.py:62]:
```suggestion
        with TestRun.step("Fill cached volume with random data."):
```
As @Kamoppl pointed out, we cannot fill cache in PT ;) Nice catch!

File: test/functional/tests/security/test_load_corrupted_metadata.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            loaded = False
+            try:
+                casadm.load_cache(cache_dev)
+                loaded = True
+                TestRun.LOGGER.info("Cache is loaded.")
+            except CmdException:
+                TestRun.LOGGER.info("Cache is not loaded.")
+
+        if loaded:
+            with TestRun.step("Run random I/O traffic to cache."):
```
Comment (Deixx) [inline test/functional/tests/security/test_load_corrupted_metadata.py:85]:
```suggestion
            with TestRun.step("Run random I/O traffic to cached volume."):
```

**Review by Deixx — APPROVED** [review]

**Review by Kamoppl — APPROVED** [review]

**Review by Deixx — APPROVED** [review]

**Pull request merged**